File: LLM-Translations/GPT-4/codenet/Java/C/s518927375.c
#include <stdio.h>

int main() {
    long k, a, b;

    scanf("%ld %ld %ld", &k, &a, &b);

    if (b <= a + 2) {
        printf("%ld\n", 1 + k);
        return 0;
    }

    long x = k - (a - 1);
    long ans = a + (x / 2) * (b - a) + (x % 2);

    printf("%ld\n", ans);
    return 0;
}

// 

File: LLM-Translations/GPT-4/codenet/Java/Go/s518927375.go
package main

import (
	"fmt"
)

func main() {
	var k, a, b int64

	fmt.Scan(&k, &a, &b)

	if b <= a+2 {
		fmt.Println(1 + k)
		return
	}

	x := k - (a - 1)
	ans := a + (x/2)*(b-a) + (x % 2)

	fmt.Println(ans)
}

// 
